funcx_endpoint/funcx_endpoint/endpoint/endpoint_manager.py

```diff
@@ -0,0 +1,508 @@
+                globus_username = msg["globus_username"]
+            except Exception as e:
+                log.error(f"Invalid server command.  ({e.__class__.__name__}) {e}")
+                self._command.ack(d_tag)
+                continue
+
+            try:
+                local_user = local_user_lookup[globus_username]
+            except Exception as e:
+                log.warning(f"Invalid or unknown user.  ({e.__class__.__name__}) {e}")
+                self._command.ack(d_tag)
+                continue
+
+            try:
+                if not (command and valid_method_name_re.match(command)):
+                    raise InvalidCommandError(f"Unknown or invalid command: {command}")
+
+                command_func = getattr(EndpointManager, command, None)
+                if not command_func:
+                    raise InvalidCommandError(f"Unknown or invalid command: {command}")
+
+                command_func(self._child_args, local_user, command_args, command_kwargs)
+                log.info(
+                    f"Command process successfully forked for '{globus_username}'"
+                    f" ('{globus_uuid}')."
+                )
+            except InvalidCommandError as err:
+                log.error(str(err))
+
+            except Exception:
+                log.exception(
+                    f"Unable to execute command: {command}\n"
+                    f"    args: {command_args}\n"
+                    f"  kwargs: {command_kwargs}"
+                )
+            finally:
+                self._command.ack(d_tag)
+
+    @staticmethod
+    def start_endpoint(
+        child_args: dict[int, tuple[int, int, str, str]],
+        local_username: str,
+        args: list[str] | None,
+        kwargs: dict | None,
+    ):
+        try:
+            pid = os.fork()
+        except Exception as err:
+            log.error(f"Unable to fork child process: {err}")
+            return
+
+        if not args:
+            args = []
+        if not kwargs:
+            kwargs = {}
+
+        ep_name = kwargs.get("name", "")  # error check is later
+        proc_args = ["funcx-endpoint", "start", ep_name, "--die-with-parent", *args]
+
+        pw_rec = pwd.getpwnam(local_username)
+        udir, uid, gid = pw_rec.pw_dir, pw_rec.pw_uid, pw_rec.pw_gid
+        uname = pw_rec.pw_name
+
+        if pid > 0:
+            proc_args_s = f"({uname}, {ep_name}) {' '.join(proc_args)}"
+            child_args[pid] = (uid, gid, local_username, proc_args_s)
+            log.info(f"Creating new user endpoint (pid: {pid}) [{proc_args_s}]")
+            return
+        pid = os.getpid()
+        os.setsid()
+
+        if not ep_name:
+            exit(os.EX_DATAERR)
+
+        exit_code = 70
+        try:
+            _soft_no, hard_no = resource.getrlimit(resource.RLIMIT_NOFILE)
+            exit_code += 1
+            os.closerange(3, hard_no)
+            exit_code += 1
+
+            env = kwargs.get("env", {})
```

Comment:
A note here about possibly passing in PATH and that's how the admin can configure where funcx-endpoint is located, as it's not obvious (though mentioned in slack comment on Feb 28)